core/state_processor.go

```diff
@@ -101,9 +101,13 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	}
 	// Fail if Shanghai not enabled and len(withdrawals) is non-zero.
 	withdrawals := block.Withdrawals()
-	if len(withdrawals) > 0 && !p.config.IsShanghai(block.Number()) {
+	if !p.config.IsShanghai(block.Number()) && withdrawals != nil {
```

Comment:
Do we actually want to stop processing block if `withdrawals` are present because we're anyways making it nil below?